dynamodb/src/main/scala/uk/gov/nationalarchives/DADynamoDBClient.scala

```diff
@@ -31,6 +31,30 @@ class DADynamoDBClient[F[_]: Async](dynamoDBClient: DynamoDbAsyncClient):
   extension [T](completableFuture: CompletableFuture[T])
     private def liftF: F[T] = Async[F].fromCompletableFuture(Async[F].pure(completableFuture))
 
+  /** Deletes a list of items with primary keys of type T from Dynamo. Unprocessed items will be retried
+    *
+    * @param tableName
+    *   The name of the table
+    * @param primaryKeys
+    *   A list of primary keys to delete from Dynamo (list can be any length; requests will be batched into groups of
+    *   25)
+    * @param format
+    *   An implicit scanamo formatter for type T
+    * @tparam T
+    *   The type of the primary keys to be deleted from Dynamo
+    * @return
+    *   The List of BatchWriteItemResponse wrapped with F[_]
```

Comment:
I initially thought, this name `Write` does not make justice now and perhaps it should be changed alongwith `WriteOrDelete` as well. 

But Looking deeper I realise that, that class is not one of ours, and imaybe "WriteOrDelete" is also fine as it explicitly mentions the dual purpose.

Looks good to me as much as I understood